addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -726,6 +727,13 @@ var SnippetEditor = Widget.extend({
                 $selectorChildren = $selectorChildren.add(self.selectorChildren[i].all());
             }
         }
+        // TODO In master, avoid referencing components from another module.
+        // Disallow dropping generated table of content outside of its snippet.
+        if (this.$target[0].classList.contains('s_table_of_content_vertical_navbar')) {
```

Comment:
Because of the "position" option, the "s_table_of_content_vertical_navbar" could be replaced by the "s_table_of_content_horizontal_navbar".. And so this fix will not work with the horizontal navabar.
But anyway, isn't the fix just disabling the move buttons? Since the "position" option allows you to make all possible moves.